homeassistant/components/schlage/entity.py

```diff
@@ -30,17 +31,19 @@ def __init__(
         )
 
     @property
-    def _lock_data(self) -> LockData:
+    def _lock_data(self) -> LockData | None:
         """Fetch the LockData from our coordinator."""
-        return self.coordinator.data.locks[self.device_id]
+        return self.coordinator.data.locks.get(self.device_id, None)
 
     @property
-    def _lock(self) -> Lock:
+    def _lock(self) -> Lock | None:
         """Fetch the Schlage lock from our coordinator."""
+        if self._lock_data is None:
+            return None
```

Comment:
Is there a moment where there are calls to these properties when the device is unavailable?

Reply:
Yes. If we get a coordinator update and the lock is removed from the account, the lock will no longer be in `self._lock_data`. So that means we need to handle the case where `self._lock_data` is `None` in the `_handle_coordinator_update()` implementation: that method is called regardless of whether the entity is available.

Reply:
Okay but then we can remove a lot of nullability checks and move them to the `_handle_coordinator_update`, because that's the only place where they are relevant

Reply:
Done. 👍🏻